src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/AdjournHearingsWindowMapping.java

```diff
@@ -0,0 +1,145 @@
+                    dateType,
+                    caseData.getCcdCaseId()
+                ));
+            }
+        }
+
+        private static LocalDate calculateDateOfAdjournedHearing(SscsCaseData caseData) {
+            String dateOrPeriod = caseData.getAdjournCaseNextHearingDateOrPeriod();
+            checkForMissingData(
+                dateOrPeriod,
+                "adjournCaseNextHearingDateOrPeriod",
+                caseData
+            );
+            if (PROVIDE_PERIOD.equalsIgnoreCase(dateOrPeriod)) {
+                checkForMissingData(
+                    caseData.getAdjournCaseNextHearingFirstAvailableDateAfterPeriod(),
+                    "adjournCaseNextHearingFirstAvailableDateAfterPeriod",
+                    caseData
+                );
+                return getDateFromNumberOfDays(caseData);
+            } else if (PROVIDE_DATE.equalsIgnoreCase(dateOrPeriod)) {
+                String dateAfterDate = caseData.getAdjournCaseNextHearingFirstAvailableDateAfterDate();
+                checkForMissingData(
+                    dateAfterDate,
+                    "adjournCaseNextHearingFirstAvailableDateAfterDate",
+                    caseData
+                );
+                LocalDate date = LocalDate.parse(dateAfterDate);
+                logAdjournmentDate(caseData, String.format("first available date after %s", date.toString()));
+                return date;
+            } else {
+                throw new IllegalArgumentException(String.format(
+                    "Unexpected adjournCaseNextHearingDateOrPeriod: %s for case id %s",
+                    dateOrPeriod,
+                    caseData.getCcdCaseId()
+                ));
+            }
+        }
+
+        @NotNull
+        private static LocalDate getDateFromNumberOfDays(SscsCaseData caseData) {
+            int numberOfDays = Integer.parseInt(
+                caseData.getAdjournCaseNextHearingFirstAvailableDateAfterPeriod().substring(0, 2));
+            LocalDate result = LocalDate.now().plusDays(numberOfDays);
+            logAdjournmentDate(caseData, String.format(
+                "first available date after %s days: %s", numberOfDays, result));
+            return result;
+        }
+
+        private static void checkForMissingData(String caseDataField, String missingField, SscsCaseData caseData) {
+            if (isBlank(caseDataField)) {
+                throw new IllegalArgumentException(
+                    missingField + " unexpectedly missing for case id " + caseData.getCcdCaseId());
+            }
+        }
+
+        private static void logAdjournmentDate(SscsCaseData caseData, String details) {
+            log.info("Case {} adjourned to {}", caseData.getCcdCaseId(), details);
+        }
+    }
+
+    public static LocalDateTime getFirstDateTimeMustBe() {
+        // TODO Adjournments - Find out how to use adjournCase data to work this out, possibly related variables:
```

Comment:
Quick query, I'm guessing this out of scope for this ticket?

Reply:
Yes - this was already here before I started, and is unrelated to the date calculation